pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -325,30 +340,30 @@ func arePublicKeysEqual(a, b []*types.JWTKey) bool {
 	return true
 }
 
-// typeX509AuthoritiesToRaw parses X509 authorities to raw certificates
-func typeX509AuthoritiesToRaw(certs []*types.X509Certificate) [][]byte {
-	var raws [][]byte
-	for _, cert := range certs {
-		raws = append(raws, cert.Asn1)
-	}
+func parseBundle(b *types.Bundle) *plugintypes.Bundle {
+	typesBundle := proto.Clone(b).(*types.Bundle)
 
-	return raws
-}
-
-// typeJWTAuthoritiesToProto parse keys to common public keys
-func typeJWTAuthoritiesToProto(keys []*types.JWTKey) []*common.PublicKey {
-	var commonKeys []*common.PublicKey
-	for _, key := range keys {
-		commonKeys = append(commonKeys, &common.PublicKey{
-			PkixBytes: key.PublicKey,
-			Kid:       key.KeyId,
-			NotAfter:  key.ExpiresAt,
+	var jwtAuthorities []*plugintypes.JWTKey
+	for _, jwtauthority := range typesBundle.JwtAuthorities {
+		jwtAuthorities = append(jwtAuthorities, &plugintypes.JWTKey{
+			PublicKey: jwtauthority.PublicKey,
+			KeyId:     jwtauthority.KeyId,
+			ExpiresAt: jwtauthority.ExpiresAt,
 		})
 	}
 
-	return commonKeys
-}
+	var x509Authorities []*plugintypes.X509Certificate
```

Comment:
Can we add a conversion function to the pkg/common/coretypes/x509certificate/plugintypes.go called `ToPluginFromAPIProto(s)` that does this?